pkg/puller/puller.go

```diff
@@ -0,0 +1,501 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package puller
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/intervalstore"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/pullsync"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/topology"
+)
+
+var (
+	bins = uint8(16)
```

Comment:
well actually not a const, since i'm altering this value in tests (to make the tests more readable), it has to be a `var` in order to do so. but it could be that it would just be nicer to inject this value through the constructor. there's a few other things that could follow